plugins/CorePluginsAdmin/Commands/InstallPlugin.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link    https://matomo.org
+ * @license https://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\CorePluginsAdmin\Commands;
+
+use Piwik\Container\StaticContainer;
+use Piwik\Plugin\ConsoleCommand;
+use Piwik\Plugin\Manager;
+use Piwik\Plugins\CorePluginsAdmin\PluginInstaller;
+use Piwik\Plugins\Marketplace\Marketplace;
+
+/**
+ * plugin:install console command.
+ */
+class InstallPlugin extends ConsoleCommand
+{
+    protected function configure(): void
+    {
+        $this->setName('plugin:install');
+        $this->setDescription('Install a plugin.');
+        $this->addOptionalArgument('plugin', 'The plugin name you want to install. Multiple plugin names can be specified separated by a space.', null, true);
+    }
+
+    protected function doExecute(): int
+    {
+        $input = $this->getInput();
+        $output = $this->getOutput();
+        $pluginManager = Manager::getInstance();
+
+		if (!Marketplace::isMarketplaceEnabled()) {
+			$output->writeln(sprintf("<error>Marketplace is not enabled, can't install plugins.</error>"));
+			return self::FAILURE;
+		}
+
+        $pluginNames = $input->getArgument('plugin');
+
+        foreach ($pluginNames as $pluginName) {
+            if ($pluginManager->isPluginInstalled($pluginName)) {
+                $output->writeln(sprintf('<comment>The plugin %s is already installed.</comment>', $pluginName));
+                continue;
+            }
+
+            try {
+                $this->installPlugin($pluginName, $pluginManager);
```

Comment:
You don't need to pass in the plugin manager instance, you can call `Manager::getInstance()` in the `installPlugin` method, it uses the static container and it will give you the existing instance. It's easier and simpler to only pass in the plugin name.

Reply:
That's definitely cleaner. What if we promote it to a class property? It's being used in 2 methods...if you don't like this idea, I can always submit it as you said

Reply:
That's why we have the static container, we don't need to store it in a property, it does that for us.

Reply:
Got it. Just fixed that and rebased the code